3rd-party/woocommerce-services.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+class WC_Services_Installer {
+
+	public function init() {
+		$this->add_error_notice();
+		$this->try_install();
+	}
+
+	public function try_install() {
+		$action = filter_input( INPUT_GET, 'wc-services-action' );
+
+		if ( 'install' === $action ) {
+			check_admin_referer( 'wc-services-install' );
+
+			$result = $this->install();
+
+			wp_safe_redirect( add_query_arg( array(
+				'wc-services-action'        => false,
```

Comment:
We can also do a `wp_get_referer()` to find out where to send them back to.

https://github.com/Automattic/jetpack/blob/764a699120ca6795dac3d264207713516200ee7b/class.jetpack-admin.php#L206